### PR TITLE
Regenerate2

### PR DESCRIPTION
**Pull request by meggart**
This is a new PR, containing @visr s new C wrapper against the julia-0.5 compatible changes. This is mostly to try to see if the tests pass on 0.5 anfd for better debugging (without the tons of warning messages)

### CODE REVIEW

**Comment by visr** [comment]
Nice, I'll close my PR.
You want to only support v0.5 right? This is not yet reflected in the `REQUIRE` file, and to test on v0.5 the `.travis.yml` and `appveyor.yml` also need to be updated.

Note that the link above still goes to my personal appveyor account. I just deleted the project there and added it under a JuliaGeo account that is not connected to a GitHub account and can be shared. I'll email you the login such that you can also restart builds etc.
https://ci.appveyor.com/project/JuliaGeo/netcdf-jl

**Comment by visr** [comment]
Ah nevermind I could just add the JuliaGeo team on Appveyor.

**Comment by meggart** [comment]
Yes, this shoudl be 0.5 only. Is there any action I should take after you added the JuliaGeo team to AppVeyor?

**Comment by visr** [comment]
No action needed. The GitHub status box now shows the correct link. If you want to restart a build on Appveyor, you can login with your GitHub account, it will see you are part of JuliaGeo, and you can go to that account page.

**Comment by visr** [comment]
Ok so on Windows both are failing, 32 bit because it cannot delete a file ([link](https://ci.appveyor.com/project/JuliaGeo/netcdf-jl/build/1.0.4/job/9ya4luu24gtx1fif#L119)).
64 bit has another issue seen [here](https://ci.appveyor.com/project/JuliaGeo/netcdf-jl/build/1.0.4/job/dncjkhlnr63dsadt#L119).

Locally (x64 Windows) I see the same problem as on AppVeyor.

**Comment by meggart** [comment]
Ok, something seems to be reall wrong on windows, I will check what happens with my old Windows machine.

**Comment by visr** [comment]
Let me know if you need help. Might be a good excuse to try out Gallium.jl.

**Comment by meggart** [comment]
Ok, I think I found the reason. It is a call to nc_put_vara_long it looks like under Windows Clong means Int32 while under Mac and Linux it is an Int64. I always assumed long would be equivalent to Int64... Do you know if Clonglong is always an alias for Int64?

**Comment by visr** [comment]
> Do you know if Clonglong is always an alias for Int64?

I think in practice it is. Long long needs to be at least 64 bit.

**Comment by meggart** [comment]
Ok, this bug seems to be fixed. However, the windows tests still fail. There is something wrong with the Win32 test and the Win64 test does not give an error message at all but still shows up as failed.

**Comment by visr** [comment]
The Win64 build is going wrong at the conda stage: [Headers are 1.8.15, library is 1.8.16](https://ci.appveyor.com/project/JuliaGeo/netcdf-jl/build/1.0.6/job/d4ka4s1x3dktvcq7#L153). Possibly the conda build is broken, we could try using the [conda-forge channel](https://github.com/visr/GDAL.jl/blob/8f8d3f22072615fd8c1eb4f8954241eede3bfc25/deps/build.jl#L29)?

**Comment by visr** [comment]
This time the Win64 build was fine..

There seems to be some kind of shared state in the `nc.var` between the files. The `vt1` var added in `test/intermediate.jl` seems to still be present when creating a new NetCDF in `test/high.jl`. Or perhaps the closing doesn't work properly.

**Comment by meggart** [comment]
This is really strange, the test file `high.jl` actually generates new temporary file names, so they should not interfer. I think there might be a problem with tempname(). I just installed julia-32bit and after deleting all jl_*.tmp files in the temporary directory ran 

``` julia
readdir(tempdir())
fn1=tempname()
readir(tempdir())
```

you can see that the file returned by `tempname()` already got created. If you run this after including `high.jl` the temporary file even looks like a NetCDF file containing these variables. 

Can you try to reproduce or do you have an idea what is going on?

**Comment by meggart** [comment]
Ok, just found out that I probably hit this bug:

https://github.com/JuliaLang/julia/issues/9053

This seems to be known, but I still have no idea why the generated file looks like an existing NetCDF file, will investigate further...

**Comment by meggart** [comment]
Ok, 32bit Windows is working now, but 64bit still gives the library error. I will try to use the conda-forge channel. 
@visr When you did the libgdal build for WinRPM, did this include libnetcdf as a dependency, so that we could use this binary instead?

**Comment by meggart** [comment]
Using conda-forge fails on both 64bit and 32bit Windows.

File: deps/deps.jl [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# This is an auto-generated file; do not edit
```
Comment (visr) [inline deps/deps.jl]: accidentally committed

**Comment by visr** [comment]
> Using conda-forge fails on both 64bit and 32bit Windows.

I see, it says: `Error: ERROR: placeholder 'C:/conda/envs/_build' too short in: hdf5-1.8.17-vc9_2`
This issue is also described in https://github.com/conda-forge/geopandas-feedstock/issues/11
And may be fixed with cond build 2.0, which was just released.
https://github.com/conda/conda-build/releases/tag/2.0.0

Regarding the WinRPM solution, @yeesian prepared it for GDAL.
https://build.opensuse.org/package/show/home:yeesian/mingw64-gdal
It does not have netcdf as a dependency currently.
But I see for instance that they did something similar here, the same user also made a GDAL cross compile with many more dependencies:
https://aur.archlinux.org/packages/mingw-w64-netcdf/

**Comment by meggart** [comment]
Ok, the WinRPM solution looks a bit too complicated to me, I would suggest to wait for conda-forge to rebuild after the bug fix. Do you think we should squash and merge anyway to get the new C API and bug fixes into master and try to solve the windows dependencies in a few weeks when conda has updated?

**Comment by visr** [comment]
It's also passing on AppVeyor now. Merge?

**Comment by meggart** [comment]
Yes, thanks for your help.

**Pull request merged**